students/eowyn/session04/revised_mailroom.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python
+
+import sys
+
+''' global variables'''
+donor_names = ["Margaret Atwood", "Fred Armisen",
+               "Heinz the Baron Krauss von Espy"]
+donations = [[300, 555], [240, 422, 1000], [1500, 2300]]
+DONORS = dict(zip(donor_names, donations))
+
+
+def mainloop():
+    ''' main interactive loop 
+    "send a thank you" "create a report" or "quit"
+    '''
+    while True:
+        answer = input("Select from one of these options: \n"
+                       "(1) Send a Thank You\n"
+                       "(2) Create a Report\n"
+                       "(3) Send letters to everyone\n"
+                       "(4) Quit\n>")
+        arg_dict = {"1": thank_you,
```

Comment:
you probably want to create the arg_dict outside of the loop -- this way, you are making a new one each time through the loop, even though it isn't changing.